package/MDAnalysis/analysis/reflectivity.py

```diff
@@ -0,0 +1,293 @@
+                        digits = []
+                    else:
+                        num = 1
+                    try:
+                        element_num_dict[element] += num
+                    except:
+                        element_num_dict[element] = num
+                letters = [char]
+    return element_num_dict
+
+
+def bc_atom(element=None, isotope_dict=None): 
+    # calculate total
+    # calculate coherent scattering length from an element and isotope dict
+    lines = TABLE_ISOTOPEDATA.splitlines()
+    for line in lines:
+        words = line.split()
+        if words and words[0] == element:
+            if isotope_dict == None: abundance = float(words[2])
+            else: abundance = isotope_dict[int(words[1])]
+            bc_r = float(words[4])
+            bc_i = float(words[5])
+            try:
+                scat_length += abundance*(bc_r + 1j*bc_i) 
+            except NameError:
+                scat_length = abundance*(bc_r + 1j*bc_i) 
+    try:
+        return scat_length
+    except NameError:
+        raise ValueError("Couldn't calculate bc for element{}".format(element))
+
+
+def element_dict_to_scat_len(element_num_dict):
+    sl = 0+0j
+    for element in element_num_dict:
+        element_name = element
+        isotope_dict = None
+        if element == 'Pr': 
+            isotope_dict = {1:1,2:0,3:0}
+            element_name = 'H'
+        if element == 'D': 
+            isotope_dict = {1:0,2:1,3:0}
+            element_name = 'H'
+        if element == 'T': 
+            isotope_dict = {1:0,2:0,3:1}
+            element_name = 'H'
+        sl += element_num_dict[element]*bc_atom(element_name,isotope_dict)
+    return sl
+
+
+def isotope_to_scat_len(isotope):
+    element_num_dict = parse_isotope(isotope)
+    sl = element_dict_to_scat_len(element_num_dict)
+    return sl
+
+
+def linear_scattering_length_density(selection, bins=10, range=None,
+                                     start=None, stop=None, step=None,
+                                     interval=1, quiet=False):
+    """
+    selection : AtomGroup
+        Selection to perform the analysis on
+    bins
+        `bins` parameter passed to numpy.histogram
+    range
+        `range` parameter passed to numpy.histogram
+    start, stop, step
+        Slice the trajectory as ``trajectory[start"stop:step]``; default
+        is to read the whole trajectory.
+    quiet
+        Print status update to the screen for every *interval* frame? [``False``]
+        - ``True``: no status updates when a new frame is processed
+        - ``False``: status update every frame (including number of atoms
+          processed, which is interesting with ``update_selection=True``)
+    interval
+        Show status update every *interval* frame [1]
+
+    Remarks
+    -------
+        If the water should be null-reflecting, simply omit it from the
+        atom selection provided.
+    """
+    if type(selection) == MDAnalysis.Universe:
```

Comment:
`AtomGroup.universe` and `universe.universe` both give a Universe, and `Universe.atoms`, and `AtomGroup.atoms` both give an AtomGroup, so you can just do
```
selection = selection.atoms
u = selection.universe
```